src/tail/log_server.rs

```diff
@@ -24,26 +27,55 @@ impl LogServer {
         }
     }
 
-    pub async fn run(self) -> Result<(), failure::Error> {
-        let service = make_service_fn(|_| async { Ok::<_, hyper::Error>(service_fn(print_logs)) });
+    pub async fn run(self, format: String) -> Result<(), failure::Error> {
```

Comment:
should take an `&str` if you aren't returning an object that `format` gets moved into

Reply:
Hmm there's some issues with lifetimes around hyper stuff using borrows, I'm going to move ownership of the string into the `LogServer` struct.